lighthouse-core/gather/gather-runner.js

```diff
@@ -256,8 +239,13 @@ class GatherRunner {
     if (recordTrace) await driver.beginTrace(settings);
 
     // Navigate.
-    await GatherRunner.loadPage(driver, passContext);
-    log.timeEnd(status);
+    try {
+      await GatherRunner.loadPage(driver, passContext);
```

Comment:
we're injecting a lot of new if/try/catch blocks into some already beefy methods, do you think there's a way we might be able to push some of these complications down a level so the busier method is still easy to parse?

Reply:
yes, if we can refactor Runner.run to easily return a (errored) LHR in it's catch. Then all the code but what's in driver.js could be removed.

Reply:
We're dropping the LHR requirement for now, so we can also drop these changes (there won't be a timing object to add this to if an error was thrown)